src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -969,7 +969,7 @@ public IEnumerable<Document> GetDocuments(DocumentsOperationContext context, IEn
             return GetDocuments(context, listOfIds, start, take, totalCount);
         }
 
-        public IEnumerable<Document> GetDocumentsForCollection(DocumentsOperationContext context, IEnumerable<Slice> ids, string collection, long start, long take, Reference<int> totalCount)
+        public IEnumerable<Document> GetDocumentsForCollection(DocumentsOperationContext context, IEnumerable<string> ids, string collection, long start, long take, Reference<int> totalCount)
```

Comment:
We get a `string` id and then convert it to `Slice`
https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Server/Documents/Queries/CollectionQueryEnumerable.cs#L609

Reply:
1. This is a huge issue, because now you are allocating a Slice, making it ToLower and then allocating another string. This should stay as Slice
2. Are you doing ToLower? This method expects lowercases IDs